src/com/ms/silverking/net/async/AsyncBase.java

```diff
@@ -263,160 +258,180 @@ public void doWork(Connection connection) {
+          //} else {
+          //  Log.warning("suspectAddressListener is null");
+          //}
         }
-        
-        @Override
-        public void doWork(Connection connection) {
-            boolean cleanWrite;
-
-            if (AsyncGlobals.debug) {
-                Log.fine("AsyncBase.Writer.doWork()");
-            }
-            cleanWrite = false;
-            try {
-                connection.writeAllPending();
-                cleanWrite = true;
-            } catch (IOException ioe) {
-                if (connection.isConnected()) {
-                    Log.logErrorWarning(ioe);
-                    // Commented out for new approach.
-                    // New approach is to simply do the disconnect as before, but not mark as suspect.
-                    // If the reconnect fails, then it will get marked.
-                    //if (suspectAddressListener != null) {
-                    //  suspectAddressListener.addSuspect(connection.getRemoteSocketAddress());
-                    //} else {
-                    //  Log.warning("suspectAddressListener is null");
-                    //}
-                }
-            } catch (Exception e) {
-                if (connection.isConnected()) {
-                    Log.logErrorWarning(e, "Unhandled exception()");
-                }
-            } finally {
-                if (!cleanWrite) {
-                    disconnect(connection, "!cleanWrite");
-                }
-            }
+      } catch (Exception e) {
+        if (connection.isConnected()) {
+          Log.logErrorWarning(e, "Unhandled exception()");
         }
-    }
-                
-    ////////////////////////////////////////////////////////////////////////////////////
-    public T newOutgoingConnection(InetSocketAddress dest, ConnectionListener listener) throws IOException, ConnectionAbsorbException {
-        SocketChannel    channel;
-        
-        channel = SocketChannel.open();
-        if (dest.isUnresolved()) {
-            Log.warning("Unresolved InetSocketAddress: "+ dest);
-            throw new ConnectException("Unresolved InetSocketAddress"+ dest.toString());
+      } finally {
+        if (!cleanWrite) {
+          disconnect(connection, "!cleanWrite");
         }
-        LWTThreadUtil.setBlocked();
+      }
+    }
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////////
+  public T newOutgoingConnection(InetSocketAddress dest, ConnectionListener listener)
+      throws IOException, ConnectionAbsorbException {
+    SocketChannel channel = null;
+    boolean connectionSuccess = false;
+
+    try {
+      channel = SocketChannel.open();
+      LWTThreadUtil.setBlocked();
+      channel.socket().connect(dest, defSocketConnectTimeout);
+      T conn = addConnection(channel, listener, false);
+      connectionSuccess = true;
+      return conn;
+    } catch (UnresolvedAddressException uae) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 338.